### ISSUE TITLE
ContainsTypeMismatch false positive w/ predicate function

### ISSUE THREAD

**Issue by levinson** [HairyFotr/linter#46]
I've found a false positive with the ContainsTypeMismatch:
```
val foo = Some("foo")
val bar = Some("bar")
```
fails with the following:
```
foo.exists(bar.contains(_))
foo.exists(bar.contains)
```
... but works with:
```
foo.exists(str => bar.contains(str))
```

**Comment by HairyFotr** [comment]
For me `foo.exists(bar.contains)` gives a false positive (because Any is inferred by the compiler instead of String - am investigating why), but the other two work as expected.

**Comment by levinson** [comment]
Ah, my bad, it seems that foo.exists(bar.contains(_)) does indeed work. Thanks for looking into the issue. :+1:
